### ISSUE TITLE
JavaFx Tutorial on Fxml some conflict with handleuserinput methods?

### ISSUE THREAD

**Issue by gweeyc** [nus-tic4002-AY2122s2/forum#2]
Hi Profs and classmates,
Found this conflict when doing the JavaFx Tutorial Fxml section:-

1) MainWindow.java (UI class controller file), the handleUserInput() uses String for input/response variables.
2) Duke.java there is also the handleUserInput() but uses Label for userText/dukeText ?

![image](https://user-images.githubusercontent.com/43993134/151689651-faac60b2-4a51-405d-826b-9d0708de5ff2.png)

The tutorial instructions are not clear how to deal with these 2 files to resolve this conflict detected also in Intellij Idea?

Thank you.
Gwee Yeu Chai

**Comment by simonjulianl** [comment]
Hi Alvin, I'm a mentor for the CS2103T module which has done the Duke project before. To resolve the conflict in the `handleUserInput` method, one of the solutions is to not convert the `userInput.getText()`  into label which is the method that is used in the  `MainWindow.java` method. 

This is because:

1. `getUserDialog` and `getDukeDialog` methods accept text of `String` type and not of `Label` type. `Java` is very strict about types.
2. using `getUserDialog` as example, `getUserDialog` method would create a new `DialogBox` 
3. Within the constructor of `DialogBox` as indicate in the line of `private DialogBox(String text, Image img)`, the constructor would assign the `Label` dialog which is a `DialogBox`'s class field using the `setText()` method. 

In summary, you should not convert the input from `String` to `Label` and should pass the `userText` to `getUserDialog` method instead. Similarly for `getDukeDialog`. I hope this clarifies the issues.

Regards,
Simon

**Comment by VimuthM** [comment]
Hi Gwee Yeu Chai,

We create the `MainWindow.java` in an effort to separate the Ui and logic aspects in the `Duke.java` file, as explained here:

![image](https://user-images.githubusercontent.com/62249192/151690586-c7156b10-dac3-4892-8b6c-bf455c3d0e73.png)

So, we move the `handleUserInput()` method from `Duke.java` to the newly created `MainWindow.java`, which means you can **remove** this method in the `Duke.java` file, so there wouldn't be a conflict. This should fix your issue.

Now to go into a bit more detail as to why it changed from `Label` -> `String`, previously we had to create Ui elements within `Duke.java`, so a `Label` element was used. But now, our `.fxml` files take care of the Ui elements. 

You can see from that we specified a `TextField` element with fx:id `userInput` and set it's `onAction` property to call the `handleUserInput()` method which we now define in the `MainWindow.java` file.

![image](https://user-images.githubusercontent.com/62249192/151690751-0760efbf-b349-4f04-9f95-7425f576e286.png)

This `TextField userInput`, which is used inside `handleUserInput()`, is made accessible within `MainWindow.java` as shown below:

![image](https://user-images.githubusercontent.com/62249192/151690863-1e473c6a-195f-406d-99d6-191d7b28966a.png)

**Comment by gweeyc** [comment]
HI guys,
             Thank you for the tips and help. Appreciate it much. I will follow up with the solutions.
